app/components/content/expander_component.rb

```diff
@@ -0,0 +1,51 @@
+module Content
+  class ExpanderComponent < ViewComponent::Base
+    attr_reader :header, :title, :text, :link_title, :link_url,
+                :background, :expanded, :classes
+
+    include ActiveSupport::Inflector
+
+    def initialize(
+      title:, text:, header: "Non-UK citizens:",
+      link_title: nil,
+      link_url: nil,
+      background: "purple",
+      expanded: false,
+      classes: nil
+    )
+      super
+
+      @header = header
+      @title = title
+      @text = text
+      @link_title = link_title&.strip&.chomp(".")
+      @link_url = link_url
+      @background = background
+      @expanded = expanded
+      @classes = classes
+
+      fail(ArgumentError, "title must be present") if title.blank?
+      fail(ArgumentError, "text must be present") if text.blank?
+      fail(ArgumentError, "background must be purple") unless %w[purple].any?(background)
```

Comment:
why do we ask the question if there's only one correct answer?

Reply:
My thinking was that in the future we might allow multiple colours. At the moment the colour defaults to purple so this will only fail if someone tries to override the default with something else.

Reply:
i think YAGNI probably applies here, but i'll leave the decision in your capable hands!